subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationWorker.java

```diff
@@ -16,16 +16,16 @@
 
 package org.gradle.internal.operations;
 
-import org.gradle.api.Action;
-
 /**
- * A worker that can run build operations of type {@link T}.
+ * A worker that can run build operations.
  *
  * Implementations must be thread-safe.
  */
-public interface BuildOperationWorker<T extends BuildOperation> extends Action<T> {
+public interface BuildOperationWorker<O extends BuildOperation> {
     /**
      * Returns a human consumable name for this tool.
      */
     String getDisplayName();
+
+    void execute(O buildOperation);
```

Comment:
At the moment, custom workers need to handle everything themselves (building description, notifying listeners etc.) - not only execution.

The `execute()` method of the DefaultBuildOperationExecutor is not used here. In fact, custom workers might use custom `BuildOperations` (i.e. not use RunnableBO or CallableBO).

We can discuss if we really want that behavior. But I think that's a different discussion.